File: app/src/main/java/com/example/dev/gymassistantv2/SettingsActivity.java
package com.example.dev.gymassistantv2;

import android.app.Activity;
import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class SettingsActivity extends Activity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        //Call super class constructor
        super.onCreate(savedInstanceState);

        //ExerciseSet content view
        setContentView(R.layout.activity_settings);

        //ExerciseSet fonts
        Typeface typeface = Typeface.createFromAsset(getAssets(),  "fonts/BlackOpsOne-Regular.ttf");

        // Set up buttons
        Button button = findViewById(R.id.buttonClearDatabase);
        button.setTypeface(typeface);
        button.setOnClickListener(this);
        button = findViewById(R.id.buttonChooseAvailableExercises);
        button.setTypeface(typeface);
        button.setOnClickListener(this);
        button = findViewById(R.id.buttonDefaultSettings);
        button.setTypeface(typeface);
        button.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {

        switch(view.getId()) {

            case R.id.buttonClearDatabase:
                DatabaseHelper db = new DatabaseHelper(getApplicationContext());
                db.truncateCompletedWorkouts();
                Toast toast = Toast.makeText(getApplicationContext(),
                        getString(R.string.history_erased_notification),
                        Toast.LENGTH_LONG);
                toast.show();
                break;

            case R.id.buttonChooseAvailableExercises:
                Intent intent = new Intent(getApplicationContext(), ChooseAvailableExercisesActivity.class);
                startActivity(intent);
                break;

            case R.id.buttonDefaultSettings:
                getApplicationContext().deleteDatabase(getString(R.string.database_name));
                db = new DatabaseHelper(getApplicationContext());
                db.populateExercisesTable(getApplicationContext());
                db.populateMuscleGroupsTable(getApplicationContext());
                toast = Toast.makeText(getApplicationContext(),
                        getString(R.string.default_settings_restored_notification),
                        Toast.LENGTH_LONG);
                toast.show();
                break;
        }
    }
}

File: app/src/main/java/com/example/dev/gymassistantv2/SettingsActivity.kt
package com.example.dev.gymassistantv2

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import com.example.dev.gymassistantv2.database.GymAssistantDatabase
import com.example.dev.gymassistantv2.dtos.UserDto

class SettingsActivity : Activity() {

    private lateinit var loggedUser: UserDto

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)


        determineLoggedUserData()
        setClearHistoryButton()
        setChooseExercisesButton()
        setResetToDefaultButton()
        setAddExerciseButton()
    }

    private fun determineLoggedUserData() {
        loggedUser = this.intent.getSerializableExtra("loggedUser") as UserDto
    }

    private fun setClearHistoryButton() {
        val buttonClearWorkoutHistory = findViewById<Button>(R.id.buttonClearWorkoutsInDatabase)
        buttonClearWorkoutHistory.setOnClickListener {
            clearWorkoutHistory()
            Toast.makeText(applicationContext, "Historia trenengów została wyczyszczona!",
                    Toast.LENGTH_LONG).show()
        }
    }

    private fun clearWorkoutHistory() {
        val dbContext = GymAssistantDatabase.getInstance(this)
        dbContext!!.workoutDao().delete(dbContext.workoutDao().getForUser(loggedUser.userId!!))
    }

    private fun setChooseExercisesButton() {
        val buttonChooseExercises = findViewById<Button>(R.id.buttonChooseAvailableExercises)
        val intentChooseExercises = Intent(this, ChooseAvailableExercisesActivity::class.java)
        buttonChooseExercises.setOnClickListener { startActivity(intentChooseExercises) }
    }

    private fun setResetToDefaultButton() {
        findViewById<Button>(R.id.buttonDefaultSettings).setOnClickListener {
            clearWorkoutHistory()
            deleteCustomExercises()
            deleteTrainerAndCharges()
            Toast.makeText(applicationContext, "Aplikacja zostałą zresetowana do ustawień domyślnych!",
                    Toast.LENGTH_LONG).show()
        }
    }

    private fun deleteCustomExercises() {
        val dbContext = GymAssistantDatabase.getInstance(this)
        dbContext!!.exerciseDao().delete(loggedUser.userId!!)
    }

    private fun deleteTrainerAndCharges() {
        val dbContext = GymAssistantDatabase.getInstance(this)
        if (loggedUser.isTrainer!!) {
            var charges = dbContext!!.userDao().getChargesForUser(loggedUser.userId!!)
            charges.forEach {
                it.trainerId = null
                dbContext!!.userDao().update(it)
            }
        } else {
            loggedUser.trainerId = null
            dbContext!!.userDao().update(loggedUser.toUser())
        }
    }

    private fun setAddExerciseButton() {
        val buttonAddExercise = findViewById<Button>(R.id.buttonAddExercise)
        val intentAddExercise = Intent(this, AddExerciseActivity::class.java)
        buttonAddExercise.setOnClickListener {
            intentAddExercise.putExtra("loggedUser", loggedUser)
            startActivity(intentAddExercise)
        }
    }

    override fun onBackPressed() {
        finish()
    }
}